src/class-config.php

```diff
@@ -504,16 +504,21 @@ protected function register_graphql_field( $type_name, $field_name, $config ) {
 
 						if ( ! empty( $value ) && is_array( $value ) ) {
 							foreach ( $value as $post_id ) {
+								$post_model  = null;
 								$post_object = get_post( $post_id );
 								if ( $post_object instanceof \WP_Post ) {
 									$post_model     = new Post( $post_object );
+								}
+								
+								$post_model = apply_filters( 'graphql_acf_relationship_post_model', $post_model, $post_id, $root, $acf_field );
```

Comment:
So, this is a bit of a band-aid. 

Any other plugin like Carbon Fields, CMB2, Metabox.io, etc, etc will have similar needs as this, and going around filtering all their fields to play nice with WooCommerce custom Models doesn't scale. 

First, I think the ACF plugin needs to take advantage of Deferred loading and use the Loader. That's on me to get updated in this repo. 

This would then mean that the Model wouldn't even be instantiated in this resolver, so this would need to be handled more centrally in the Loader level.

Perhaps a central filter somewhere in the PostObjectLoader might help us re-route posts of the "product" type to the Product Model 🤔 (maybe that's already even happening?)

Then I think it might solve some headaches if WPGraphQL for WooCommerce extended the WPGraphQL Model Layer. 

From what I can tell, you extend `Model` class here: https://github.com/wp-graphql/wp-graphql-woocommerce/blob/develop/includes/model/class-crud-cpt.php#L20

And since the `Crud_CPT` Model is intended to deal with CPTs, so you should probably extend the `Post` model instead of just Model. 

This would then mean that any field expecting a `Post` model would work with any instance of a `Product` Model, because they'd be an extension of the `Post` model. 

That would, I believe, solve a lot of one-off headaches around this stuff?

Reply:
I could try that, but it could definitely break some things.